### PR TITLE
Add data attrs to (xarray) resample output

### LINKED ISSUE #40: Add (more) attributes to xarray `resample` output
As xarray uses the structure of netCDF files, there is quite a lot of space to add extra information, in the form of "attributes".

For provenance and traceability, we can make use of this in Lilio. This can consist of, but is not limited to:

- Adding attributes to all coordinates to explain what they are (e.g. anchor_year). [left_bound and right_bound currently do have a description in the attrs](https://github.com/AI4S2S/lilio/blob/d0923997b4fd8cdb7ea41712006f6055ae7deb0d/lilio/utils.py#L167-L174). 
- Adding the calendar itself to the xr.Dataset/xr.DataArray main attrs.

### PR DESCRIPTION
**Pull request by BSchilperoort**
Closes #40

Several attributes are added to the xarray-data output of `resample`, for both clarity and provenance.

### CODE REVIEW

**Comment by BSchilperoort** [comment]
I saw that pandas also [supports adding attributes to `DataFrame`s](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.attrs.html), however this features is marked as "experimental" and can change without warning, so it might still be too early to support this.

File: lilio/_attrs.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+            f"(here: {calendar.anchor})."
+        ),
+    }
+    data["i_interval"].attrs = {
+        "name": "interval index",
+        "units": "-",
+        "description": (
+            "The index of each Lilio Calendar interval. Positive values denote "
+            "intervals after the anchor date, while negative values represent intervals"
+            " before the anchor date."
```
Comment (geek-yang) [inline lilio/_attrs.py:43]:
```suggestion
            "The index of each Lilio Calendar interval. Positive values denote "
            "intervals after the anchor date (targets), while negative values "
            "represent intervals before the anchor date (precursors)."
```
To remind the user that this is consistent with the defined precursors and targets when creating the calendar object.

**Review by geek-yang — APPROVED** [review]
Good job!

**Comment by BSchilperoort** [comment]
Thanks for the review and suggestion!

**Pull request merged**